samza-core/src/main/scala/org/apache/samza/checkpoint/OffsetManager.scala

```diff
@@ -245,14 +263,12 @@ class OffsetManager(
 
       // invoke checkpoint listeners
       //partitionOffsets.groupBy(_._1.getSystem).foreach {
-      partitionOffsets.groupBy { case (ssp, _) => ssp.getSystem }.foreach {
+      checkpoint.getOffsets.asScala.groupBy { case (ssp, _) => ssp.getSystem }.foreach {
         case (systemName:String, offsets: Map[SystemStreamPartition, String]) => {
```

Comment:
Nitpick: Unrelated, Since checkpoint.offsets is already a java.util.Map, I think this is doing unnecessary Java -> Scala -> Java conversion. Can you try making this `offsets: util.Map[SSP, String]` and remove the `offsets.asJava` below and see if it still compiles/works?